src/network/networkd-link.c

```diff
@@ -1511,6 +1511,15 @@ static int link_up(Link *link) {
         }
 
         if (link->network->mtu) {
+                /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes
+                   on the interface. Bump up MTU bytes to IPV6_MTU_MIN. */
+                if (socket_ipv6_is_supported() && link->network->mtu < IPV6_MIN_MTU) {
```

Comment:
Meh. I don't think this is a good choice. If I want to setup a low-MTU IPIP tunnel or TUN/TAP, which has absolutely nothing to do with IPv6, the MTU will still be bumped to 1280 because socket_ipv6_is_supported() would be true, as in the current version. That is not so desirable.